contrib/IECoreArnold/src/IECoreArnold/MeshAlgo.cpp

```diff
@@ -265,37 +185,25 @@ AtNode *convertCommon( const IECore::MeshPrimitive *mesh )
 	variablesToConvert.erase( "P" ); // These will be converted
 	variablesToConvert.erase( "N" ); // outside of this function.
 
-	// Convert and remove indexed UVs first. We must perform
-	// the iteration to find the names separately to the iteration
-	// to convert them, because convertIndexedUVSet() removes items
-	// from variablesToConvert, and would therefore invalidate
-	// the interators we were using if we were to do it in one loop.
-	vector<string> uvSetNames;
-	for( PrimitiveVariableMap::iterator it = variablesToConvert.begin(), eIt = variablesToConvert.end(); it != eIt; ++it )
-	{
-		if( boost::ends_with( it->first, "Indices" ) )
-		{
-			uvSetNames.push_back( it->first == "stIndices" ? "" : it->first.substr( 0, it->first.size() - 7 ) );
-		}
-	}
-	for( vector<string>::const_iterator it = uvSetNames.begin(), eIt = uvSetNames.end(); it != eIt; ++it )
-	{
-		convertIndexedUVSet( *it, variablesToConvert, result );
-	}
-
-	// Then convert and remove non-indexed uvs. As above, we must
-	// do this in two phases.
-	uvSetNames.clear();
+	// Find all UV sets. We must perform the iteration to find the
+	// names separately to the iteration to convert them, because
+	// convertUVSet() removes items from variablesToConvert, and
```

Comment:
convertUVSet doesn't remove items any more...

Reply:
Oops.. you're right. Ignore my comment above where I claimed that it did...

Reply:
Pretty sure I did try condensing that down to one loop and still had issues though.

Reply:
I think it can be condensed provided you don't use `it` after `erase( it )`. I think the C++11 idiom is this, because erase now returns the next iterator, whereas before it didn't return anything :

```
for( auto it = c.begin(), it != c(); /* note we don't increment here */ )
{
    if( wantToErase( it ) )
    {
       it = c.erase( it );
    }
    else
    {
        ++it;
    }
}
```

Reply:
Addressed in 214ec0781bc75d1f48bd413abcd407f9cb56e69c